config/config.go

```diff
@@ -59,10 +72,58 @@ func Init(flags *pflag.FlagSet) {
 		if flag.Name == "help" {
 			return
 		}
-		viper.Set(flag.Name, flag.Value.String())
+
+		// Only updated parsed configs if the user sets value
+		if flag.Changed {
+			viper.Set(flag.Name, flag.Value.String())
+		}
 	})
 
 	if err := viper.Unmarshal(&Config); err != nil {
 		panic(err)
 	}
 }
+
+// ConfigureParentDirPaths Creates the default parent directory which can be used for logs and persistent data
+func configureParentDirPaths() {
+	var err error
+	DefaultParentDir, err = createDefaultParentDir()
+	if err != nil {
+		slog.Warn("Failed to create default preferences directory", slog.String("error", err.Error()))
+	}
+}
+
+func createDefaultParentDir() (string, error) {
+	// Get the default directory path for the OS
+	prefDir := GetDefaultParentDirPath()
+	if err := os.MkdirAll(prefDir, os.ModePerm); err != nil {
+		return "", fmt.Errorf("failed to create preferences directory: %w", err)
+	}
+
+	return prefDir, nil
+}
+
+func GetDefaultParentDirPath() string {
+	switch runtime.GOOS {
+	case "linux":
+		return linuxPath
+	case "darwin":
+		return expandHomeDirDarwin(darwinPath)
+	case "windows":
+		return windowsPath
+	default:
+		slog.Warn("unsupported OS, defaulting to current directory")
+		return DefaultParentDir
+	}
+}
+
+func expandHomeDirDarwin(path string) string {
+	if path != "" && path[0] == '~' {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			return path // Fallback if home directory can't be resolved
+		}
+		return filepath.Join(home, path[1:])
+	}
+	return path
+}
```

Comment:
Again, this might not be required. Let's have absolute paths only, the more we support this, the more are the edge cases that we will need tp handle.